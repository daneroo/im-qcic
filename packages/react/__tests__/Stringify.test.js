import React from 'react'
import renderer from 'react-test-renderer'

import { Stringify } from '../src/index.js'

describe('Stringify', () => {
  test('import Stringify', () => {
    expect(typeof Stringify).toBe('function')
  })
  test('Stringify renders', () => {
    const component = renderer.create(
      <Stringify />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Stringify renders array', () => {
    const component = renderer.create(
      <Stringify data={[1, 2, 3]} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Stringify renders nested Object', () => {
    const component = renderer.create(
      <Stringify data={{ this: 'one', that: 'other', those: ['extra', 'things'] }} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
