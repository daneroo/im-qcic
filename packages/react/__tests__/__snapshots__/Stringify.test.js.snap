// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stringify Stringify renders 1`] = `
<div>
  <pre
    style={
      Object {
        "fontSize": "70%",
        "textAlign": "left",
      }
    }
  />
</div>
`;

exports[`Stringify Stringify renders array 1`] = `
<div>
  <pre
    style={
      Object {
        "fontSize": "70%",
        "textAlign": "left",
      }
    }
  >
    [
  1,
  2,
  3
]
  </pre>
</div>
`;

exports[`Stringify Stringify renders nested Object 1`] = `
<div>
  <pre
    style={
      Object {
        "fontSize": "70%",
        "textAlign": "left",
      }
    }
  >
    {
  "this": "one",
  "that": "other",
  "those": [
    "extra",
    "things"
  ]
}
  </pre>
</div>
`;
